modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authorization/authorizer/rdbms/resource/dao/impl/AuthResourceInMemoryDao.java

```diff
@@ -0,0 +1,117 @@
+import java.util.stream.Collectors;
+
+/**
+ * Class implements {@link AuthResourceDao} to provide functionality manage in memory auth resources.
+ */
+public class AuthResourceInMemoryDao implements AuthResourceDao {
+
+    /**
+     * In memory resource store for non durable resources like exchanges, queues in a table structure as,
+     * 1. resource type
+     * 2. resource name
+     * 3. auth resource
+     */
+    private Table<String, String, AuthResource> inMemoryResourceMap = HashBasedTable.create();
+
+    @Override
+    public void persist(AuthResource authResource) throws BrokerAuthServerException {
+        inMemoryResourceMap.put(authResource.getResourceType(),
+                                authResource.getResourceName(),
+                                authResource);
+    }
+
+    @Override
+    public void update(AuthResource authResource) throws BrokerAuthServerException {
+        inMemoryResourceMap.put(authResource.getResourceType(),
+                                authResource.getResourceName(),
+                                authResource);
+    }
+
+    @Override
+    public void delete(String resourceType, String resource) throws BrokerAuthServerException {
+        inMemoryResourceMap.remove(resourceType, resource);
+    }
+
+    @Override
+    public AuthResource read(String resourceType, String resource) throws BrokerAuthServerException {
+        return inMemoryResourceMap.get(resourceType, resource);
+    }
+
+    @Override
+    public List<AuthResource> readAll(String resourceType, String ownerId) throws BrokerAuthServerException {
+        Map<String, AuthResource> resourceMap = inMemoryResourceMap.row(resourceType);
+        if (Objects.nonNull(resourceMap)) {
+            Collection<AuthResource> authResources = resourceMap.values();
+            return authResources.stream()
+                                .filter(authResource -> authResource.getOwner()
+                                                                    .equals(ownerId))
+                                .collect(Collectors.toList());
+        }
+        return null;
+    }
+
+    @Override
+    public List<AuthResource> readAll(String resourceType, String action, String ownerId, List<String> userGroups)
+            throws BrokerAuthServerException {
+        Map<String, AuthResource> resourceMap = inMemoryResourceMap.row(resourceType);
+        if (Objects.nonNull(resourceMap)) {
+            Collection<AuthResource> authResources = resourceMap.values();
+            return authResources.stream()
+                                .filter(authResource -> authResource.getOwner().equals(ownerId) ||
+                                        checkActionAndGroups(authResource.getActionsUserGroupsMap(),
+                                                             action,
+                                                             userGroups))
+                                .collect(Collectors.toList());
+        }
+        return null;
```

Comment:
Can we return an empty list instead of null?

Reply:
Fixed in 885868b